mailet/standard/src/main/java/org/apache/james/transport/mailets/LogMessage.java

```diff
@@ -52,18 +52,81 @@
      * or mark it as finished.
      */
     private final Logger logger;
+    private final LogLevel logLevel;
     private boolean passThrough = true;
-    private boolean headers = true;
-    private boolean body = true;
-    private int bodyMax = 0;
-    private String comment = null;
+    private boolean headers;
+    private boolean body;
+    private int bodyMax;
+    private String comment;
+    private List<String> specificHeaders;
+    private List<String> specificAttributes;
+
+    private StringBuffer logBuffer = new StringBuffer();
```

Comment:
Please consider that mailet are concurrently called. Your code needs to be safe in a multi-threaded context.

Hence sharing a buffer seems 'unsafe'